config/kernel-setattr-prepare.m4

```diff
@@ -0,0 +1,17 @@
+dnl #
+dnl # 4.9 API change,
+dnl # inode_change_ok is changed to setattr_prepare and takes dentry
+dnl #
+AC_DEFUN([ZFS_AC_KERNEL_SETATTR_PREPARE], [
+	AC_MSG_CHECKING([whether setattr_prepare() exists])
+	ZFS_LINUX_TRY_COMPILE([
+		#include <linux/fs.h>
+	],[
+		setattr_prepare(NULL, NULL);
+	],[
+		AC_MSG_RESULT(yes)
+		AC_DEFINE(HAVE_SETATTR_PREPARE, 1, [setattr_prepare() exists])
+	],[
+		AC_MSG_RESULT(no)
+	])
+])
```

Comment:
Let's steal my configure check, https://github.com/zfsonlinux/zfs/pull/5301/commits/3d8c8e8fa16636c8a038a5a6742cc9a76c2388a6, for this since it also verifies the argument types to `setattr_prepare` which means if they change we'll notice immediately.